File: backend-dotnet/Jerrygram.Api/Models/Post.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Jerrygram.Api.Models
{
    public class Post
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        /// <summary>
        /// Azure Blob URL
        /// </summary>
        [MaxLength(300)]
        public string? ImageUrl { get; set; } = string.Empty;

        [MaxLength(2200)]
        public string? Caption { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // FK
        [Required]
        public Guid UserId { get; set; }

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        public ICollection<Comment> Comments { get; set; } = [];
        public ICollection<PostLike> Likes { get; set; } = [];

        /// <summary>
        /// Visibility of the post (Public, FollowersOnly, Private)
        /// </summary>
        [Required]
        public PostVisibility Visibility { get; set; } = PostVisibility.Public;
    }
}

File: jerrygram-recommend/models/Post.js
export class Post {
  constructor({
    id,
    caption,
    imageUrl,
    createdAt,
    likes = 0,
    liked = false,
    user,
    score = 0
  }) {
    this.id = id;
    this.caption = caption;
    this.imageUrl = imageUrl;
    this.createdAt = createdAt;
    this.likes = likes;
    this.liked = liked;
    this.user = user;
    this.score = score;
  }

  static fromDbRow(row) {
    return new Post({
      id: row.Id,
      caption: row.Caption,
      imageUrl: row.ImageUrl,
      createdAt: row.CreatedAt,
      likes: parseInt(row.Likes, 10),
      liked: false,
      user: {
        id: row.UserId,
        username: row.Username,
        profileImageUrl: row.ProfileImageUrl
      }
    });
  }

  toJSON() {
    return {
      id: this.id,
      caption: this.caption,
      imageUrl: this.imageUrl,
      createdAt: this.createdAt,
      likes: this.likes,
      liked: this.liked,
      user: this.user,
      ...(this.score > 0 && { score: this.score })
    };
  }
}